parser/src/main/scala/hmda/parser/fi/ts/TsCsvParser.scala

```diff
@@ -1,95 +1,71 @@
-        val respCity = values(9)
-        val respState = values(10)
-        val respZip = values(11)
-        val parentName = values(12)
-        val parentAddress = values(13)
-        val parentCity = values(14)
-        val parentState = values(15)
-        val parentZip = values(16)
-        val contactPerson = values(17)
-        val contactPhone = values(18)
-        val contactFax = values(19)
-        val contactEmail = values(20)
+    val id = toIntOrFail(values(0), "Record Identifier")
+    val respId = values(1)
+    val code = toIntOrFail(values(2), "Agency Code")
+    val timestamp = toLongOrFail(values(3), "Timestamp")
+    val activityYear = toIntOrFail(values(4), "Activity Year")
+    val taxId = values(5)
+    val totalLines = toIntOrFail(values(6), "Total Lines")
+    val respName = values(7)
+    val respAddress = values(8)
+    val respCity = values(9)
+    val respState = values(10)
+    val respZip = values(11)
+    val parentName = values(12)
+    val parentAddress = values(13)
+    val parentCity = values(14)
+    val parentState = values(15)
+    val parentZip = values(16)
+    val contactPerson = values(17)
+    val contactPhone = values(18)
+    val contactFax = values(19)
+    val contactEmail = values(20)
 
-        val respondent = Respondent(respId, respName, respAddress, respCity, respState, respZip)
-        val parent = Parent(parentName, parentAddress, parentCity, parentState, parentZip)
-        val contact = Contact(contactPerson, contactPhone, contactFax, contactEmail)
-        Right(
-          TransmittalSheet(
-            id,
-            code,
-            timestamp,
-            activityYear,
-            taxId,
-            totalLines,
-            respondent,
-            parent,
-            contact
-          )
-        )
-      }
-      case scalaz.Failure(errors) => {
-        Left(
-          errors.list.toList
-        )
-      }
-    }
-  }
+    val respondent = Respondent(respId, respName, respAddress, respCity, respState, respZip)
+    val parent = Parent(parentName, parentAddress, parentCity, parentState, parentZip)
+    val contact = Contact(contactPerson, contactPhone, contactFax, contactEmail)
 
-  def checkTs(fields: List[String]): ValidationNel[String, List[AnyVal]] = {
+    val maybeTS: ValidationNel[String, TransmittalSheet] = (
+      id
+      |@| code
+      |@| timestamp
+      |@| activityYear
+      |@| taxId.success
+      |@| totalLines
+      |@| respondent.success
+      |@| parent.success
+      |@| contact.success
+    ) { TransmittalSheet }
 
-    if (fields.length != 21) {
-      ("Incorrect number of fields. found: " + fields.length + ", expected: 21").failure.toValidationNel
-    } else {
-      val numericFields = ListMap(
-        "Record Identifier" -> fields(0),
-        "Agency Code" -> fields(2),
-        "Activity Year" -> fields(4),
-        "Total Lines Entries" -> fields(6)
-      )
+    maybeTS.disjunction.toEither.bimap(_.toList, identity(_))
```

Comment:
Are you planning to change this line in this PR, or should I go ahead and merge?